plugins/inputs/modbus/configuration_register.go

```diff
@@ -216,7 +218,7 @@ func (c *ConfigurationOriginal) validateFieldDefinitions(fieldDefs []fieldDefini
 
 			// check scale
 			if item.Scale == 0.0 {
-				return fmt.Errorf("invalid scale '%f' in '%s' - '%s'", item.Scale, registerType, item.Name)
+				// return fmt.Errorf("invalid scale '%f' in '%s' - '%s'", item.Scale, registerType, item.Name)
```

Comment:
Ummmm... why?

Reply:
My understanding is that if the scale is not specified, go sets it to 0.0 but the user probably had 1.0 in mind. Is there another way to distinguish the case where the user actively wrote 0.0 or did not set the field at all?

Reply:
Well in the old style it was mandatory to specify `scale`. You can change that, but silently accepting `0.0` which will then throw away data is not the right way. Either handle that default in the `init...(` functions or keep the error message that prevents data loss.